### PR TITLE
Report sort error

### PR DESCRIPTION
**Pull request by pflanze**
resolves "also reject sorting" request on Slack.

### Summary

IIRC we never discussed the further plans for sorting when streaming. Probably refusing with an error ("unimplemented") is really the right thing to do so far. It's a bit weird and potentially problematic in that FASTA/FASTA aligned queries will succeed with sorting first, then when the data set grows until the result set becomes >10'000 rows, they will start to fail. But that may still be better than silently wrong sorting. It would be good to warn the user in advance (while there are still <10'000 rows) somehow; again something that NDJSON cannot deliver, while JSON could contain richer data than just the rows. Documentation will have to be enough, or perhaps the dashboard or other UI components could warn about such queries.
I worry a bit more now that Details should also be changed to stream, and will expose the same conundrum.

As per discussion (Fabian), for Details we might set the cut-off a lot higher though since/if the involved data is much smaller, and might make the problem go away for most organisms.

There is no test as it's still impossible to test without large data or future work, thus will test manually.

User documentation should be added, but where?

## PR Checklist

- [ ] All necessary documentation has been adapted or there is an issue to do so.
~~- [ ] The implemented feature is covered by an appropriate test.~~

### CODE REVIEW

**Comment by Taepper** [comment]
I thought the second part of streaming was now to introduce the config cutoff option?

**Comment by pflanze** [comment]
> I thought the second part of streaming was now to introduce the config cutoff option?

This is not "the second part of streaming", but the second part of implementing the (small) followup requests for the first part.

"The second part of streaming" will contain the parts that make it possible to automatically test.

**Comment by pflanze** [comment]
I forgot to update the comment in the code that it's not a hack anymore, will do later, have to leave.

File: src/silo_api/query_handler.cpp [hunk 1]
```diff
@@ -50,9 +50,11 @@ void QueryHandler::post(
             );
          }
       }
-   } catch (const silo::QueryParseException& ex) {
+   } catch (const silo::QueryException& ex) {
       response.setContentType("application/json");
-      SPDLOG_INFO("Query is invalid: " + query + " - exception: " + ex.what());
+      SPDLOG_INFO(
+         "Query is invalid: {} - exception during {}: ", query, ex.duringString(), ex.what()
```
Comment (fengelniederhammer) [inline src/silo_api/query_handler.cpp:56]:
```suggestion
         "Query is invalid: {} - exception during {}: {}", query, ex.duringString(), ex.what()
```

**Review by fengelniederhammer — APPROVED** [review]
LGTM

**Comment by fengelniederhammer** [comment]
I think we can make the `fix: action: use more appropriate new QueryEvaluationException` a `refactor`, since it's internal only.

File: src/silo_api/query_handler.cpp (hunk 1 above)
Comment (pflanze) [reply src/silo_api/query_handler.cpp:56]: Oh, no compile time check. (Another reason to use Rust.) Good catch!

Comment (fengelniederhammer) [reply src/silo_api/query_handler.cpp]: I was wondering why the log statement didn't include the message... :D

**Pull request merged**